src/Tizen.Peripheral/Tizen.Peripheral/Uart.cs

```diff
@@ -0,0 +1,333 @@
+        None = 0,
+
+        /// <summary>Even parity is used.</summary>
+        Even,
+
+        /// <summary>Odd parity is used.</summary>
+        Odd
+    }
+
+    /// <summary>
+    /// Enumeration for stop bits.
+    /// </summary>
+    public enum StopBit
+    {
+        /// <summary>One stop bit.</summary>
+        Bits1Bit = 0,
+
+        /// <summary>Two stop bits.</summary>
+        Bits2Bit
+    }
+
+    /// <summary>
+    /// Enumeration for hardware flow control.
+    /// </summary>
+    public enum HardwareFlowControl
+    {
+        /// <summary>No hardware flow control.</summary>
+        None = 0,
+
+        /// <summary>Automatic RTS/CTS hardware flow control.</summary>
+        AutoRtsCts
+    }
+
+    /// <summary>
+    /// Enumeration for software flow control.
+    /// </summary>
+    public enum SoftwareFlowControl
+    {
+        /// <summary>No software flow control.</summary>
+        None = 0,
+
+        /// <summary>XON/XOFF software flow control.</summary>
+        XonXoff,
+    }
+
+    /// <summary>
+    /// The class allows applications to communicate via UART platform's bus.
+    /// </summary>
+    /// <privilege>http://tizen.org/privilege/peripheralio</privilege>
+    public class Uart : IDisposable
+    {
+        private BaudRate _baudRate;
+        private ByteSize _byteSize;
+        private Parity _parity;
+        private StopBit _stopBits;
+        private HardwareFlowControl _hardwareFlowControl;
+        private SoftwareFlowControl _softwareFlowControl;
+
+        /// <summary>
+        /// Native handle to I2c.
+        /// </summary>
+        protected IntPtr _handle;
+
+        private Uart(IntPtr handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Opens the UART slave device.
+        /// </summary>
+        /// <param name="port">The UART port number that the slave device is connected.</param>
+        /// <returns>The UART object.</returns>
+        public static Uart Open(int port)
+        {
+            var ret = NativeUart.Open(port, out IntPtr handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return new Uart(handle);
+        }
+
+        /// <summary>
+        /// Closes the UART slave device.
+        /// </summary>
+        public void Close()
+        {
+            var ret = NativeUart.Close(_handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Disposes Uart object.
+        /// </summary>
+        public void Dispose()
+        {
+            Close();
+        }
+
+        /// <summary>
+        /// Reads the bytes data from the UART slave device.
+        /// </summary>
+        /// <param name="size">The size of array to read.</param>
+        /// <returns>The array of bytes.</returns>
+        public byte[] Read(int size)
```

Comment:
The form of using the buffer inside the `read` method prevents the reuse of the buffer.
Also, when an Exception occurs, the created Buffer is just discarded.
You also need a `Read` method that takes the Buffer itself as a method argument.